### PR TITLE
Move EVP functions to evp.c

### PR DESCRIPTION
**Pull request by julek-wolfssl**

### CODE REVIEW

File: tests/api.c [hunk 1]
```diff
@@ -22186,7 +22186,7 @@ static void test_wolfSSL_CTX_add_client_CA(void)
 
 static void test_wolfSSL_X509_NID(void)
 {
-    #if (defined(OPENSSL_EXTRA) || defined(OPENSSL_EXTRA_X509_SMALL)) && \
+    #if defined(OPENSSL_EXTRA) && \
```
Comment (JacobBarthelmeh) [inline tests/api.c:22189]: We need to keep the same functionality for --enable-opensslextra=x509small as before.

With these changes wolfSSL_X509_get_pubkey, PKEY_new/free, and init are missing from the build. Comparison of size of the compiled code looks good though.

File: src/ssl.c [hunk 2]
```diff
@@ -145,6 +140,9 @@
 #endif
 #endif /* !WOLFCRYPT_ONLY || OPENSSL_EXTRA */
 
+#define WOLFSSL_EVP_INCLUDED
+#include "wolfcrypt/src/evp.c"
```
Comment (JacobBarthelmeh) [inline src/ssl.c:144]: With the include remaining here the build process for evp.c is not separated. That's ok, and may have not been the goal of this PR, just placing this comment incase it was part of the goal.

File: src/ssl.c [hunk 3]
```diff
@@ -34816,7 +30693,7 @@ static int SetECPointInternal(WOLFSSL_EC_POINT *p)
 #endif /* HAVE_ECC */
 #endif /* OPENSSL_EXTRA */
 
-#if defined(HAVE_ECC) && defined(OPENSSL_EXTRA_X509_SMALL)
+#if defined(HAVE_ECC) && defined(OPENSSL_EXTRA)
```
Comment (JacobBarthelmeh) [inline src/ssl.c:30696]: This used to be a feature that was a subset of OPENSSL_EXTRA code. For a use case that was looking for a subset of X509 operations pulled from the OPENSSL_EXTRA build.

File: tests/api.c (hunk 1 above)
Comment (julek-wolfssl) [reply tests/api.c]: The missing functions should be back in the OPENSSL_EXTRA_X509_SMALL build.

**Review by JacobBarthelmeh — APPROVED** [review]

**Review by toddouska — APPROVED** [review]

**Pull request merged**